### PR TITLE
Add common function for user agent

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
I realize the line is a little vague here in terms of "Is this UI related" for the "ui" package, but it shares some code with 'report an issue' and every azure extension needs it. I'm just saying I'm okay with it being here, but let me know if you really care otherwise.

### CODE REVIEW

File: ui/index.d.ts [hunk 1]
```diff
@@ -630,3 +630,8 @@ export interface UIExtensionVariables {
     ui: IAzureUserInput;
     reporter: TelemetryReporter | undefined;
 }
+
+/**
+ * Adds a user agent specific to the VS Code extension, of the form `${extensionName}/${extensionVersion}`
+ */
+export declare function addExtensionUserAgentInfo(client: ServiceClient): void;
```
Comment (StephenWeatherford) [inline ui/index.d.ts]: nit: I'd say the "Info" part is unnecessary

File: ui/src/getPackageInfo.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    // tslint:disable-next-line:strict-boolean-expressions
+    const extensionName: string = (packageJson && packageJson.name) || 'vscode-azuretools';
+    // tslint:disable-next-line:strict-boolean-expressions
+    const extensionVersion: string = (packageJson && packageJson.version) || 'Unknown';
+    return [extensionName, extensionVersion];
+}
+
+interface IPackageJson {
+    version?: string;
```
Comment (StephenWeatherford) [inline ui/src/getPackageInfo.ts]: Why are these fields optional?

File: ui/src/getPackageInfo.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import { ext } from "./extensionVariables";
+
+export function getPackageInfo(): [string, string] {
+    let packageJson: IPackageJson | undefined;
+    try {
+        // tslint:disable-next-line:non-literal-require
+        packageJson = <IPackageJson>require(ext.context.asAbsolutePath('package.json'));
+    } catch (error) {
+        // ignore errors
```
Comment (StephenWeatherford) [inline ui/src/getPackageInfo.ts]: telemetry on error?

File: ui/index.d.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply ui/index.d.ts]: Fine with me. I was just being consistent with `client.addUserAgentInfo`

**Pull request merged**